python/ngen_cal/src/ngen/cal/ngen_hooks/ngen_output.py

```diff
@@ -0,0 +1,95 @@
+from __future__ import annotations
+
+from ngen.cal import hookimpl
+
+import pandas as pd
+from pandas import DataFrame, Series
+from pathlib import Path
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from ngen.cal.meta import JobMeta
+
+from typing import TYPE_CHECKING
+
+if TYPE_CHECKING:
+    from ngen.cal.meta import JobMeta
+
+class TrouteOutput:
+
+    def __init__(self, filepath: Path) -> None:
+        self._output_file = filepath
```

Comment:
Thoughts on doing something like:

```python
assert filepath.exist(), f"filepath: {str(filepath)} does not exits"
assert filepath.suffix.lower() in (".csv", ".hdf5"), f"unsupported file type: {filepath.suffix}"
```

I think then we could clean up the return type of `_get_dataframe` to just be a `DataFrame`. To me, that just makes the invariants really clear.

Reply:
We use the `None` type to indicate to the application that an output doesn't exist so it should attempt to run the model to produce an initial output...

Reply:
Fair enough. I didn't think this would run before a model ever would.

Reply:
> "We" being me

😆